coretex/cli/modules/utils.py

```diff
@@ -17,20 +17,50 @@
 
 from typing import List, Any, Optional, Callable
 from functools import wraps
+from importlib.metadata import version as getLibraryVersion
 
 import sys
 
 from py3nvml import py3nvml
 
 import click
+import requests
 
+from . import ui
 from ...utils.process import command
 
 
 def updateLib() -> None:
     command([sys.executable, "-m", "pip", "install", "--no-cache-dir", "--upgrade", "coretex"], ignoreStdout = True, ignoreStderr = True)
 
 
+def fetchLatestVersion() -> Optional[str]:
+    url = "https://pypi.org/pypi/coretex/json"
+    response = requests.get(url)
+
+    if response.ok:
+        data = response.json()
+        return str(data.get("info", None).get("version", None))
```

Comment:
`data.get` will return `None` if "info" key is not present which will raise an error. Properly validate that all these values are contained inside of a dictionary and that the are proper types.